### PR TITLE
"Contributing"

### PR DESCRIPTION
**Pull request by arokem**
This file gets rendered by Github into PRs (https://github.com/blog/1184-contributing-guidelines) and should help newcomers figure out what/how/why to contribute. For the git procedure, I am directing them to the existing developer documentation, but this also covers some other points that we like to look for in our review. Am I forgetting anything important?

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Code contributions should be formatted according to the
+[PEP8](https://www.python.org/dev/peps/pep-0008/) style guide. There are
+[software systems](https://pypi.python.org/pypi/pep8) that will check your code
+for PEP8 compliance, and most text editors can be configured to check the
+compliance of your code with PEP8.
+
+### Documentation
+
+We use [sphinx](http://www.sphinx-doc.org/en/stable/contents.html) for
+documentation. We welocme contributions of examples, and suggestions for
```
Comment (jchoude) [inline CONTRIBUTING]: Typo in welcome

File: CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+changes in the documentation, but please make sure that changes that are
+introduced render properly into the html format that is used for the Dipy
+website
```
Comment (jchoude) [inline CONTRIBUTING]: Missing the closing period.

File: doc/devel/python3.rst [hunk 3]
```diff
@@ -25,22 +28,22 @@ As the first code line of the file, to use Python 3 behavior by default.
 Print
 *****
 
-Yeah, you knew that, but use the ``__future__`` import above, and
-``print(something)``
+In Python 3, ``print`` is  function. Please use the ``__future__`` import above,
```
Comment (jchoude) [inline doc/devel/python3.rst]: Missing an "a" between "is" and "function"?

**Comment by jchoude** [comment]
Seems like a good addition. Very nice to know that they now render this automatically!

**Comment by arokem** [comment]
Thanks for taking a look! I fixed that typos you identified and also added
a sentence about the numpy docstring standard, which I forgot to mention
previously.

File: CONTRIBUTING [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Contributing to Dipy
+
+Dipy is an open-source software project, and we have an open development
+process. This means that we welcome contributions from anyone. We do ask that
+you first read this document and follow the guidelines we have outlined here and
+that you follow the [NIPY community code of conduct](http://nipy.org/conduct.html).
```
Comment (Garyfallidis) [inline CONTRIBUTING]: Is this conduct inspired from somewhere or written from scratch?

File: CONTRIBUTING [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+that are not run in the tests. You can also see the test coverage in the Travis
+run corresponding to the PR (in the log for the machine with ``COVERAGE=1``).
+
+Contributions to tests that extend test coverage in older modules that are not
+fully covered are very welcome!
+
+### Supporting both Python 2 and 3
+
+Most of the functionality in Dipy works on both Python 3 and Python 2. Please
+follow the instructions [here](http://nipy.org/dipy/devel/python3.html) to
```
Comment (Garyfallidis) [inline CONTRIBUTING]: You can write http://dipy.org/devel/...
We got the domain so we can make the links shorter. However, it's a bit puzzling that when someone writes dipy.org/devel we will get nipy.org/dipy/devel in his browser but maybe there is a way to fix this (from godaddy).

File: CONTRIBUTING [hunk 6]
```diff
@@ -0,0 +1,89 @@
+# Contributing to Dipy
```
Comment (Garyfallidis) [inline CONTRIBUTING]: Thanx for starting to write such a document. Shouldn't this be an rst file in the dipy/doc folder? 
Here you wrote a markdown file but we don't have any markdown documents in dipy. I have a feeling this is in the wrong format and possibly folder. This information should  be rendered by sphinx or something in the docs. You can also have a copy in the setup.py folder but that is secondary.

File: CONTRIBUTING (hunk 4 above)
Comment (arokem) [inline CONTRIBUTING]: As it says, inspired by others and adapted for our needs. See https://github.com/nipy/nipy.github.com/pull/1

**Comment by arokem** [comment]
It's a markdown file, and at top level, so that it interoperates well with Github. See here: https://github.com/blog/1184-contributing-guidelines. TLDR: If there's a markdown file called `CONTRIBUTING.md` (or `CONTRIBUTING`) at top level, people get pointed to this file when they make a PR.

File: CONTRIBUTING (hunk 6 above)
Comment (arokem) [inline CONTRIBUTING]: Fixed that other things as well. Thanks for taking a look!

**Pull request merged**